terralab/utils.py

```diff
@@ -248,3 +257,27 @@ def format_timestamp(timestamp_string: str | None) -> str:
         local_timezone
     )
     return datetime_obj.strftime("%Y-%m-%d %H:%M:%S %Z")
+
+
+# This is a workaround to make the retry messages from urllib3 more user-friendly.
+# Previously they would show up as ugly errors, i.e.
+#   Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None))
+#   after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection
+#   object at 0x102c47ef0>: Failed to establish a new connection: [Errno 61]
+#   Connection refused')': /api/pipelineruns/v1/pipelineruns?limit=10
+
+
+class RetryMessageFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        # Matches the retry warning messages produced by urllib3
+        if (
+            "Retrying (" in record.getMessage()
+            and "after connection broken by" in record.getMessage()
+        ):
+            record.msg = "terralab encountered a problem connecting to the server. Retrying your request..."
+            record.args = ()
+        return True
```

Comment:
What does the boolean represent here?  Is it always `True` because if we returned `False` it would remove the log from being emitted entirely?

Reply:
Yup, that's exactly right